### ISSUE TITLE
ssh connection denied

### ISSUE THREAD

**Issue by SimonLab** [dwyl/gogs-server#2]
I've created a new account on https://gogs-server.fly.dev/ and added my existing public ssh key to my new account:

- check if I have existing ssh keys on my machine: `ls ~/.ssh` show `id_rsa` and `id_rsa.pub` and `known_hosts` files.
- copy the public key `cat id_rsa.pub` then copy paste the content https://gogs-server.fly.dev/user/settings/ssh

- I've created a new organisation and repository and try to clone it with `git clone ssh://git@gogs-server.fly.dev:10022/theorg/myrepo.git`
However I have the following error:
![image](https://user-images.githubusercontent.com/6057298/167426787-711d2048-5106-4009-a332-2e97d050b4aa.png)

I'm looking at creating new keys (without breaking the existing one) and to try to setup the new private/public key to be used wit the gogs server only. I think I might need to create a config file to select the correct ssh key depending on the host.

**Comment by SimonLab** [comment]
Server restarted however still have the same error.
There are also maybe a related error on the log:
```sh
2022-05-09T14:51:41Z app[67a73f8c] lhr [info]May  9 14:51:41 sshd[602]: error: kex_exchange_identification: Connection closed by remote host
2022-05-09T14:51:41Z app[67a73f8c] lhr [info]May  9 14:51:41 sshd[602]: Connection closed by 205.234.240.68 port 55212
```

**Comment by nelsonic** [comment]
TCP port `55212` is not related to what you're doing. unless that's your IP address ... 💭

**Comment by SimonLab** [comment]
The errors keeps appearing but on different ports each time, https://fly.io/apps/gogs-server/logs

**Comment by nelsonic** [comment]
I'm surprised that Fly.io isn't blocking this traffic at their Firewall. It's very clearly a scan. 🙃

**Comment by nelsonic** [comment]
FYI: SSH is still working after the server restart. I just pushed to the repo you created:
https://gogs-server.fly.dev/theorg/myrepo
![image](https://user-images.githubusercontent.com/194400/167439214-d17302ba-5635-4184-a28b-301216f21cb0.png)

**Comment by SimonLab** [comment]
Good idea I'll try with another user on my machine

**Comment by nelsonic** [comment]
@SimonLab can you try:

```sh
ssh -T ssh://git@gogs-server.fly.dev:10022
```
You should see:

```sh
Hi there, You've successfully authenticated, but Gogs does not provide shell access.
```

e.g:

<img width="907" alt="image" src="https://user-images.githubusercontent.com/194400/167441870-650d4575-ddd7-42d6-8dad-84bbac64b2ba.png">

This is the baseline test for SSH.

As per: https://docs.github.com/en/authentication/connecting-to-github-with-ssh/testing-your-ssh-connection

**Comment by SimonLab** [comment]
Yes I also tried this earlier, and got `git@gogs-server.fly.dev: Permission denied (publickey,keyboard-interactive).`

**Comment by nelsonic** [comment]
Ah. OK so it's _probably_ the RSA Key algo? let's stay on the 16:30 call and debug if you have time.

**Comment by nelsonic** [comment]
@SimonLab any luck? LMK if you have time to pair either today or tomorrow morning.

**Comment by nelsonic** [comment]
@SimonLab please login to the gogs-server:

```sh
fly ssh console
```
And check that your `public` key is in this file:

```sh
/data/git/.ssh/authorized_keys
```

**Comment by nelsonic** [comment]
See: https://www.ssh.com/academy/ssh/authorized_keys/openssh#format-of-the-authorized-keys-file

**Comment by nelsonic** [comment]
Try just coping all the data on the first line of the file into your local text editor, replace my `public` key with yours,
then insert it as a new line in the file. 💭

**Comment by SimonLab** [comment]
Gogs didn't add my ssh key automatically I had to run the admin command:
![image](https://user-images.githubusercontent.com/6057298/167601533-4f1c3100-db98-4a86-95ea-dd0b7970f664.png)
Checked the `/data/git/.ssh/authorized_keys` file and my key is now there.
testing `ssh -T ssh://git@gogs-server.fly.dev:10022` returns:

>Hi there, You've successfully authenticated, but Gogs does not provide shell access.
If this is unexpected, please log in with password and setup Gogs under another user.

@nelsonic It might be worth checking on your side you can still access gogs-server with your ssh key as the file of authorized keys was rewritten

**Comment by SimonLab** [comment]
I'm going to try the solution describe on the gogs issue above:

- in `conf/app.ini` set `START_SSH_SERVER` to false
- redeploy application with flyctl
- Remove my current ssh key and add another one to see if authorized_keys file is automatically updated.

**Comment by SimonLab** [comment]
`START_SSH_SERVER` to false didn't seem to work.
When removing a ssh key the `authorized_keys` file is updated, however add a new key doesn't update it automatically and I had to run the `rewrite` command (https://github.com/dwyl/gogs-server/issues/2#issuecomment-1122176417) to update the file, then I'm able to use ssh.
I don't think it's a too big issue at the moment because we'll only have the Phoenix server talking to the gogs-server via ssh at the start

**Comment by nelsonic** [comment]
Yeah, that's my thinking. We will add SSH keys to the "dev" server for new team members to be able to run the app on their `localhost` without having to run a local `Gogs` instance. But in terms of `staging` and `production` it should really only be done _once_. 💭

**Comment by SimonLab** [comment]
Adding new key via the gogs server interface works now. I was updating the wrong `conf/app.ini` file instead of updating the configuration file directly on the fly server (see https://github.com/dwyl/gogs-server#intialize-gogs)
So setting `START_SSH_SERVER` to `false` works!
